extensions/amp-mustache/0.1/amp-mustache.js

```diff
@@ -33,14 +36,37 @@ export class AmpMustache extends AMP.BaseTemplate {
 
   /** @override */
   compileCallback() {
+    /** @private @const {!JsonObject} */
+    this.nestedTemplates_ = dict();
+    const shadowTemplate = this.element.cloneNode(true);
```

Comment:
This means we're making 2 copies of everything in the element, once here and once in `templateContent`.

Reply:
Right, in the 2nd case of templateContent where HTMLTemplateElement isn't supported natively. I changed templateContent to always return a clone (renamed to templateContentClone) and modified it so that it won't make a 2nd copy here. Also got rid of the "shadow" from the name that you didn't like ;)